tests/legacy-cli/e2e_runner.ts

```diff
@@ -175,13 +175,22 @@ Promise.all([findFreePort(), findFreePort()])
       await runSteps(runSetup, allSetups, 'setup');
       await runSteps(runInitializer, allInitializers, 'initializer');
       await runSteps(runTest, testsToRun, 'test');
-
+    } finally {
+      registryProcess.kill();
+      secureRegistryProcess.kill();
+    }
+  })
+  .then(
+    () => {
       if (shardId !== null) {
         console.log(colors.green(`Done shard ${shardId} of ${nbShards}.`));
       } else {
         console.log(colors.green('Done.'));
       }
-    } catch (err) {
+
+      process.exit(0);
```

Comment:
Why setting the exit code is needed? (If needed it would be better to use `process.exitCode` instead.)